### PR TITLE
Add LaTex requirements in README

### PR DESCRIPTION
**Pull request by xinye1**
#110 

New section **LaTeX for version `2.0.0+`** Under section **Installation**

Edit: also the Rmd is knitted to generate the md.

### CODE REVIEW

File: README.Rmd [hunk 1]
```diff
@@ -90,6 +90,19 @@ install.packages("slackr")
...
+For versions `2.0.0+` (on Windows 10 and Ubuntu 20.04 as tested), if on loading the package you see error messages like below:
+
+```{r latex, eval=FALSE}
+#> Error: package or namespace load failed for ‘slackr’:
+#> .onLoad failed in loadNamespace() for 'texPreview', details:
+#>  call: FUN(X[[i]], ...)
+#>  error: neither tlmgr or mpm are installed and in %PATH%
+```
+
+Try installing a version of LaTeX (e.g. MikTeX or Tex Live) on your system, and depending on the LaTeX version you may see warnings about various missing packages for `texPreview`, but they can be ignored.
```
Comment (mrkaye97) [inline README.Rmd]: Can we add MacTex to the list for Mac users, just to be more explicit?

File: README.Rmd [hunk 2]
```diff
@@ -90,6 +90,19 @@ install.packages("slackr")
 devtools::install_github("mrkaye97/slackr")
 ```
 
+#### LaTeX for version `2.0.0+`
+
+For versions `2.0.0+` (on Windows 10 and Ubuntu 20.04 as tested), if on loading the package you see error messages like below:
```
Comment (mrkaye97) [inline README.Rmd]: And likely on MacOS as well

**Review by mrkaye97 — CHANGES_REQUESTED** [review]
Nice work! Thanks for putting in this PR. If you don't mind spending the time, I think that adding a test to check if there's a LaTeX install (in the tests dir) makes sense. I can look into that if you don't want to. I'd also just add a tidbit about MacOS / MacTex to the LaTeX section too, just to be explicit about this potentially being an issue on MacOS too. Thanks again for these changes!

**Comment by xinye1** [comment]
> Nice work! Thanks for putting in this PR. If you don't mind spending the time, I think that adding a test to check if there's a LaTeX install (in the tests dir) makes sense. I can look into that if you don't want to. I'd also just add a tidbit about MacOS / MacTex to the LaTeX section too, just to be explicit about this potentially being an issue on MacOS too. Thanks again for these changes!

A test is a good idea, I'll have a look

Edit: on a second thought, as this is a `texPreview` issue, wouldn't it make more sense to fix it there? @yonicd I'd appreciate your thoughts. Thanks

**Review by mrkaye97 — APPROVED** [review]

**Comment by yonicd** [comment]
Requirements of texPreview are listed in the texPreview [readme](https://github.com/yonicd/texPreview#functionality), you can point users to that.

If users need more info there is OS specific setups in the gha workflow ymls eg [macos](https://github.com/yonicd/texPreview/blob/master/.github/workflows/R-mac.yml#L46)

  - __System Requirements__: 
    - Must have pdflatex in PATH, Windows users can install by running [installr::install.MikTeX](http://talgalili.github.io/installr/reference/install.MikTeX.html)
    - TeX libraries that are used: `standalone`, `xcolor`, `booktabs`, `multirow`, `array`, `helvet`, `amsmath`, `rotating`, `listings`, `graphicx`, `setspace`, `caption`
      - To Check if the system is in compliance with these libraries run `texPreview::check_requirements()`.
    - Ghostscript: If you are installing TeX with [tinytex](https://github.com/yihui/tinytex) make sure to install ghostscript onto your system.

**Comment by mrkaye97** [comment]
> Requirements of texPreview are listed in the texPreview [readme](https://github.com/yonicd/texPreview#functionality), you can point users to that.
> 
> If users need more info there is OS specific setups in the gha workflow ymls eg [macos](https://github.com/yonicd/texPreview/blob/master/.github/workflows/R-mac.yml#L46)
> 
> * **System Requirements**:
>   
>   * Must have pdflatex in PATH, Windows users can install by running [installr::install.MikTeX](http://talgalili.github.io/installr/reference/install.MikTeX.html)
>   * TeX libraries that are used: `standalone`, `xcolor`, `booktabs`, `multirow`, `array`, `helvet`, `amsmath`, `rotating`, `listings`, `graphicx`, `setspace`, `caption`
>     
>     * To Check if the system is in compliance with these libraries run `texPreview::check_requirements()`.
>   * Ghostscript: If you are installing TeX with [tinytex](https://github.com/yihui/tinytex) make sure to install ghostscript onto your system.

Ah, thanks for this. Didn't notice the texPreview readme. Seems like we can just add a test pretty easily to check `texPreview::check_requirements()`?

**Comment by xinye1** [comment]
Thanks @yonicd, the gha lines are exactly what people need. I will add the ref to README

@mrkaye97 may I suggest a different approach? As TeX is only needed for `tex_slackr` (e.g. someone like me is unlikely to ever use it), it may be better to be lazy-loaded, e.g. only prompt the user to install / load `texPreview` when `tex_slackr` is called.

**Comment by mrkaye97** [comment]
> Thanks @yonicd, the gha lines are exactly what people need. I will add the ref to README
> 
> @mrkaye97 may I suggest a different approach? As TeX is only needed for `tex_slackr` (e.g. someone like me is unlikely to ever use it), it may be better to be lazy-loaded, e.g. only prompt the user to install / load `texPreview` when `tex_slackr` is called.

Sounds like a good approach to me! I don't have strong feelings either way, and I think lazy loading makes sense.

**Comment by yonicd** [comment]
lazy loading sounds like a good option

**Pull request closed without merging**